src/prefect/engine/serializers/base.py

```diff
@@ -0,0 +1,71 @@
+import base64
+import json
+from typing import Any, Union
+
+import cloudpickle
+
+
+class Serializer:
+    """
+    Serializers are used by Results to handle the transformation of
+    Python objects to and from bytes.
+    """
+
+    def __eq__(self, other: Any) -> bool:
+        return type(self) == type(other)
+
+    def serialize(self, value: Any) -> bytes:
+        """
+        Serialize an object to bytes.
+
+        Args:
+            - value (Any): the value to serialize
+
+        Returns:
+            - bytes: the serialized value
+        """
+        pickled = cloudpickle.dumps(value)
+        return base64.urlsafe_b64encode(pickled)
```

Comment:
Just habit, I think. I'll revert to the generic function.

Reply:
Sorry, why are we base64 encoding at all?

Reply:
Ah, if there's a specific motivation for b64'ing the pickle I'm not aware - I was just moving the existing logic from the `Result` class into this `Serializer` (which is why I thought you were asking why I changed the b64 to be url_safe). I'm open to removing it and just returning the pickled bytes - @cicdw is there a historical reason we shouldn't?

Reply:
Because flows that create results will always be run in the same environment (with the same version of Prefect), I don't think we have any backwards compatibility concern about changing this.

Reply:
Hmmm the only reason I can think of would be if for some reason S3 / GCS have a base64 encoding constraint?  I can't see anything like that in the docs but it might be worth running a quick sanity check against those backends.

Reply:
For values to store in the bucket? There is no restriction in bytes stored in a bucket.

Reply:
I had an impluse it was for location as well (hence urlsafe), but it's just the bytes payload. I think I'm going to remove the b64.

Reply:
Is there a time where the serializer might load a value that was serialized with the old version? If so, we may want to try unpickling directly, and fallback to base64decode + unpickle for backwards compatibility.

Reply:
Under normal use, no, but that's not a good enough guarantee. Added a fallback.